addons/spreadsheet/static/src/pivot/pivot_runtime.js

```diff
@@ -0,0 +1,137 @@
+/** @odoo-module */
+
+import { _t } from "@web/core/l10n/translation";
+import { PERIODS, parseGroupField } from "@spreadsheet/pivot/pivot_helpers";
+
+class PivotRuntime {
+    constructor(rawDefinition, fields) {
+        this._sortedColumn = rawDefinition.sortedColumn;
+
+        this._measures = rawDefinition.measures.map((name) => new PivotMeasure(fields, name));
+        this._columns = rawDefinition.colGroupBys.map((name) => new PivotDimension(fields, name));
+        this._rows = rawDefinition.rowGroupBys.map((name) => new PivotDimension(fields, name));
+    }
+
+    get definition() {
+        return {
+            sortedColumn: this._sortedColumn,
+            measures: this._measures,
+            columns: this._columns,
+            rows: this._rows,
+        };
+    }
+}
+
+export class OdooPivotDataSource extends PivotRuntime {
```

Comment:
Isn't it supposed to be named 
```suggestion
export class OdooPivotRuntime extends PivotRuntime {
```
Or I am  missing something ?